File: 参考/comp9041-Q/cs9041/lab06/whale.py
#!/usr/bin/python3
from collections import defaultdict
import re,sys
a = defaultdict(list)
while True:
	s = sys.stdin.readline()
	if not s:
		break
	match = re.match(r'^(\d+) (.*)$',s)
	try:
		num = match.group(1)
		# print(num)
		name = match.group(2)
	except:
		pass
	a[name].append(int(num))
print("{} observations: {} pods, {} individuals"\
	.format(sys.argv[1],len(a[sys.argv[1]]),sum(a[sys.argv[1]])))

File: 参考/comp9041-Q/cs9041/lab06/whale.pl
#!/usr/bin/perl -w
while ($input = <STDIN>){
	chomp $input;
	$input =~ /(\d+) (.*)/;
	if (not $data{$2}){
		$data{$2} = "$1";
	}
	else{
		$data{$2} = "$data{$2},$1"
	}
}

$target = shift @ARGV or die;

if ($data{$target}) {
	@arr = split(',',$data{$target});
	$sum=eval(join("+",@arr));
	$pod=$#arr+1;
	print "$target observations: $pod pods, $sum individuals","\n";
}
else{
	print "$target observations: 0 pods, 0 individuals","\n";
}
